git-tar/function/ops_test.go

```diff
@@ -100,3 +102,106 @@ func Test_getRepositoryURL_whenGetTokenReturnsError_WhenRepositoryIsPrivate(t *t
 		t.Errorf("Expected URL: %s, Got: %s", expectedURL, gotURL)
 	}
 }
+
+func Test_formatTemplateReposValid(t *testing.T) {
+	formalRepos := []string{
+		"https://github.com/openfaas/templates",
+		"https://github.com/openfaas-incubator/node8-express-template.git",
+		"https://github.com/openfaas-incubator/golang-http-template.git",
+	}
+
+	tests := []struct {
+		title         string
+		envRepos      string
+		expectedRepos []string
+	}{
+		{
+			title:         "Templates with no added custom repositories",
+			envRepos:      "",
+			expectedRepos: formalRepos,
+		},
+		{
+			title:         "Templates with single added custom repository",
+			envRepos:      "https://github.com/my-custom/repo.git",
+			expectedRepos: append(formalRepos, "https://github.com/my-custom/repo.git"),
+		},
+		{
+			title:         "Templates with two added custom repositories without spaces",
+			envRepos:      "https://github.com/my-custom/repo.git,https://github.com/another/repo.git",
+			expectedRepos: append(formalRepos, ([]string{"https://github.com/my-custom/repo.git", "https://github.com/another/repo.git"})...),
+		},
+	}
+	for _, test := range tests {
+		os.Setenv("custom_templates", test.envRepos)
+		t.Run(test.title, func(t *testing.T) {
```

Comment:
This level of nesting is confusing. I think we can do better this will help with maintainability.